enumerable_methods.rb

```diff
@@ -0,0 +1,63 @@
+# Project 2: Enumerable Methods
+
+module Enumerable
+  def my_each(&proc)
+    length.times do |i|
+      proc.call(self[i])
+    end
+    self
+  end
+
+  def my_each_with_index(&proc)
+    length.times do |item, _i|
+      proc.call(item)
+    end
+    self
+  end
+
+  def my_select(&proc)
+    selection = []
+    length.times do |i|
+      selection.push(self[i]) if proc.call(self[i])
+    end
+    selection
+  end
+
+  def my_all?(&proc)
+    length.times do |i|
+      return false if proc.call(self[i]) == false
+    end
+  end
```

Comment:
* [ ] should have working `my_all?` identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_all? returns true if the block never returns false or nil
  ```ruby
    array = [1,2,5,4,7]
    true_block = proc { |x| x < 10 }
    p array.my_all?(&true_block) == array.all?(&true_block) # false
  ```

* [ ] my_all? when no block or argument is given returns true when none of the collection members are false or nil
  ```ruby
    array = [1,2,5,4,7]
    p array.my_all? == array.all? # false
  ```
* [ ] my_all? when a class is passed as an argument returns true if all of the collection is a member of such class
  ```ruby
    array = [1,2,5,4,7]
    p array.my_all?(Integer) == array.all?(Integer) # false
  ```
* [ ] my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
  ```ruby
    array = %w[dog door rod blade]
    p array.my_all?(/c/) == array.all?(/c/) # false
  ```
* [ ] my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
  ```ruby
    array = %w[dog door rod blade]
    p array.my_all?(5) == array.all?(5) # false
  ```